python/tank/platform/software_launcher.py

```diff
@@ -0,0 +1,406 @@
+# Copyright (c) 2016 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+"""
+Defines the base class for DCC application launchers all Tank engines
+should implement.
+"""
+
+import os
+import sys
+import re
+import logging
```

Comment:
Same for the `re` module import. Go through and make sure all of the imports are needed and ditch any you're not using.